protocols/imap/src/main/java/org/apache/james/imap/message/request/GetAnnotationRequest.java

```diff
@@ -0,0 +1,105 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.message.request;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+import org.apache.james.imap.api.ImapCommand;
+
+import com.google.common.base.Preconditions;
+
+public class GetAnnotationRequest extends AbstractImapRequest {
+    private final String mailboxName;
+    private Set<String> keys;
+
+    private long maxsizeValue;
+
+    public void setDepthValue(Depth depthValue) {
+        this.depthValue = depthValue;
+    }
+
+    private Depth depthValue;
+
+    public GetAnnotationRequest(String tag, ImapCommand command, String mailboxName) {
+        super(tag, command);
+        Preconditions.checkNotNull(mailboxName);
+        this.mailboxName = mailboxName;
+        this.depthValue = Depth.ZERO;
+        this.keys = ImmutableSet.of();
+    }
+
+    public boolean isValidParams() {
+        return (depthValue == Depth.ZERO || (depthValue != Depth.ZERO && !keys.isEmpty()));
```

Comment:
The boolean logic in it is non obvious. Maybe you can extract some logic from it to make it easier ?